File: SoloLearn Codes/Hofstadter's Q-Sequence/HofstadtersQSequence.cs
using System;

namespace SoloLearn
{
    class Program
    {
        static int[] t = new int[10000];

        static int q(int n)
        {
            if (n > 2)
            {
                if (t[n] == 0) t[n] = q(n - q(n - 1)) + q(n - q(n - 2));
                return t[n];
            }
            else return 1;
        }

        static void Main(string[] args)
        {
            Console.WriteLine(q(Convert.ToInt32(Console.ReadLine())));
        }
    }
}

File: SoloLearn Codes/Hofstadter's Q-Sequence/hofstadters_q_sequence.swift
var t = [Int] (repeating: 0, count: 10000)

func q(n: Int) -> Int {
    if n > 2 {
        if t[n] == 0 {
            t[n] = q(n: n - q(n: n - 1)) + q(n: n - q(n: n - 2))
        }      
        return t[n]
    } else {
        return 1
    }
}

let x = Int(readLine()!)!
print(q(n: x))